modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/SchemaHolderImpl.java

```diff
@@ -387,8 +408,11 @@ private void rebuild() {
         SchemaPlus newCalciteSchema = Frameworks.createRootSchema(false);
         newCalciteSchema.add("UUID", typeFactory -> ((IgniteTypeFactory)typeFactory).createCustomType(UUID.class));
         newCalciteSchema.add("OTHER", typeFactory -> ((IgniteTypeFactory)typeFactory).createCustomType(Object.class));
-        newCalciteSchema.add("PUBLIC", new IgniteSchema("PUBLIC"));
-        igniteSchemas.forEach(newCalciteSchema::add);
+        newCalciteSchema.add(QueryUtils.DFLT_SCHEMA, new IgniteSchema(QueryUtils.DFLT_SCHEMA));
+
+        for (IgniteSchema schema : igniteSchemas.values())
```

Comment:
I like similar to the previous: `igniteSchemas.values().forEach(sch->sch.register(newCalciteSchema));`